### PR TITLE
get launches from cache

### PR DESCRIPTION
**Pull request by VivianLi98**
Launches will populate from cache if available. I would like some opinions on this though: The current implementation always populates from cache if it is available, but should launches populate from API on app start up instead? 

EDIT: Celina made a slack thread about this.

### CODE REVIEW

File: app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt [hunk 1]
```diff
@@ -1,57 +1,129 @@
...
+    @ApplicationContext private val context: Context
 ){
 
     suspend fun getLaunches() : List<Launch?>? {
-        Log.d("Repository", "attempting to fetch launch data")
+
+        val cachedLaunches = getLaunchesFromCache()
+        if(cachedLaunches != null){
+            Log.d("Repository", "get launch from cache")
+            return cachedLaunches
```
Comment (chrsbell) [inline app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt:31]: This is interesting, I hadn't considered caching launches this way. Is there an advantage to using SharedPreferences as a cache vs checking for already inserted launches from the room database and returning those? Overall looks/works well! Just deciding on how or if we should update the cache I guess?

File: app/src/androidTest/java/com/example/voyagerx/database/SimpleEntityReadWriteTest.kt [hunk 2]
```diff
@@ -78,4 +81,16 @@ class SimpleEntityReadWriteTest {
         assertThat(user, equalTo(null))
     }
 
+    @Test
```
Comment (chrsbell) [inline app/src/androidTest/java/com/example/voyagerx/database/SimpleEntityReadWriteTest.kt:84]: Love the tests!

File: app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt (hunk 1 above)
Comment (VivianLi98) [reply app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt:31]: I had initially thought about using the values from the database, but I was having difficulty implementing it and its still a fairly expensive operation to query from the db. I think this way is faster (?).

Comment (chrsbell) [reply app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt:31]: Good point, I guess in theory we could benchmark the difference if this were a real production app but hey it's a cache. This should make the landing page much quicker, thank you!

**Pull request merged**